src/Package/PackageDumper.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Package;
+
+use App\Entity\Package;
+use App\Entity\PackageRepository;
+use App\Entity\User;
+use App\Model\DownloadManager;
+use App\Model\FavoriteManager;
+use Doctrine\ORM\EntityManagerInterface;
+use Generator;
+use function Safe\fclose;
+use function Safe\fopen;
+use function Safe\fwrite;
+use function Safe\realpath;
+use function Safe\rename;
+use function Safe\tempnam;
+
+/**
+ * Dumps all packages, together with basic information, as a big JSON file, to the web root.
+ */
+class PackageDumper
+{
+    private EntityManagerInterface $em;
+
+    private DownloadManager $downloadManager;
+
+    private FavoriteManager $favoriteManager;
+
+    private string $webDir;
+
+    public function __construct(
+        EntityManagerInterface $em,
+        DownloadManager $downloadManager,
+        FavoriteManager $favoriteManager,
+        string $webDir
+    ) {
+        $this->em              = $em;
+        $this->downloadManager = $downloadManager;
+        $this->favoriteManager = $favoriteManager;
+        $this->webDir          = realpath($webDir);
+    }
+
+    public function dump(): void
+    {
+        $tmpFile = tempnam(sys_get_temp_dir(), 'packages-full.json.');
+
+        $fp = fopen($tmpFile, 'wb');
+        fwrite($fp, '[');
+
+        $comma = '';
+
+        foreach ($this->generatePackageData() as $packageData) {
+            fwrite($fp, $comma . json_encode($packageData, JSON_PRETTY_PRINT));
+            $comma = ',';
```

Comment:
shouldn't this be `",\n"` if you want the output to be pretty-printed ?

Reply:
It depends on whether you want it to look like:

```
[{
    ...
},{
    ...
}]
```

or:

```
[{
    ...
},
{
    ...
}]
```

or even:

```
[
{
    ...
},
{
    ...
}
]
```